src/Microsoft.ApplicationInsights.AspNetCore/Implementation/TelemetryConfigurationOptionsSetup.cs

```diff
@@ -119,6 +121,18 @@ private void AddTelemetryChannelAndProcessors(TelemetryConfiguration configurati
                 {
                     configuration.TelemetryProcessorChainBuilder.UseAdaptiveSampling();
                 }
+
+                // Disable heartbeat if user sets it (by default it is on)
+                if (!this.applicationInsightsServiceOptions.EnableHeartbeat)
+                {
+                    foreach (var module in TelemetryModules.Instance.Modules)
```

Comment:
Will this singleton be initialized at all in aspnetcore case?

Reply:
Yes. I'm adding tests to prove that now.

Reply:
Unfortunately ,no from my experiments and this issue
https://github.com/Microsoft/ApplicationInsights-aspnetcore/issues/506

Reply:
Just lucky(?) that only modules added from base sdk is present. (DiagnosticsTelemetryModule is added by base sdk)

Reply:
Yes, this is just lucky and we are indeed relying on the implementation details in the base SDK here.

Tests that I added are failing outside of the Visual Studio environment, I'm investigating now.